tests/unit/Domain/Package/PackageTest.php

```diff
@@ -0,0 +1,98 @@
+<?php 
+declare(strict_types = 1);
+
+namespace PackageHealth\PHP\Test\Unit\Domain\Package;
+
+use PHPUnit\Framework\TestCase;
+
+use PackageHealth\PHP\Domain\Package\Package;
+
+final class PackageTest extends TestCase
+{
+  private Package $package;
+  private \DateTimeImmutable $createdAt;
+  
+  public function setUp(): void
+  {
+    $this->package = new Package(
+      name:           'vendor/project',
+      description:    'An awesome package description',
+      latestVersion:  '1.0',
+      url:            'http://'
+    );
+    
+    $this->createdAt = $this->package->getCreatedAt();
+  }
+  
+  public function testName(): void
+  {
+    $this->assertEquals('vendor/project', $this->package->getName());
+  }
+  
+  public function testVendor(): void
+  {
+    $this->assertEquals('vendor', $this->package->getVendor());
+  }
+  
+  public function testProject(): void
+  {
+    $this->assertEquals('project', $this->package->getProject());
+  }
+  
+  public function testDescription(): void
+  {
+    $this->assertEquals('An awesome package description', $this->package->getDescription());
+  }
+  
+  public function testDescriptionHasBeenUpdated(): void
+  {
+    $this->assertInstanceOf(\DateTimeImmutable::class, $this->package->getCreatedAt());
+    $this->assertFalse($this->package->isDirty());
+    $this->assertNull($this->package->getUpdatedAt());
+    
+    $this->package = $this->package->withDescription('Even awesomer description');
+    
+    $this->assertTrue($this->package->isDirty());
+    $this->assertEquals('Even awesomer description', $this->package->getDescription());
+    $this->assertEquals($this->createdAt, $this->package->getCreatedAt());
+    $this->assertInstanceOf(\DateTimeImmutable::class, $this->package->getUpdatedAt());
+  }
+  
+  public function testLatestVersion(): void
+  {
+    $this->assertEquals('1.0', $this->package->getLatestVersion());
+  }
+  
+  public function testLatestversionHasBeenUpdated(): void
+  {
+    $this->assertInstanceOf(\DateTimeImmutable::class, $this->package->getCreatedAt());
+    $this->assertFalse($this->package->isDirty());
+    $this->assertNull($this->package->getUpdatedAt());
+    
+    $this->package = $this->package->withLatestVersion('1.0.1');
+    
+    $this->assertTrue($this->package->isDirty());
+    $this->assertEquals('1.0.1', $this->package->getLatestVersion());
+    $this->assertEquals($this->createdAt, $this->package->getCreatedAt());
+    $this->assertInstanceOf(\DateTimeImmutable::class, $this->package->getUpdatedAt());
```

Comment:
Prefer adding `use DateTimeImmutable` to the file header and removing the leading `\`